mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,125 @@
+from azure.storage.blob import BlobClient
+
+import os
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.utils.string_utils import strip_suffix
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+from mlflow.utils.rest_utils import call_endpoint, extract_api_info_for_service
+from mlflow.protos.databricks_artifacts_pb2 import DatabricksMlflowArtifactsService
+from mlflow.protos.databricks_artifacts_pb2 import GetCredentialsForWrite, GetCredentialsForRead
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.protos.service_pb2 import MlflowService, ListArtifacts
+from mlflow.utils.uri import extract_and_normalize_path
+from mlflow.utils.proto_json_utils import message_to_json
+
+_PATH_PREFIX = "/api/2.0"
+
+
+class DatabricksArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure/AWS with access control.
+
+    The artifact_uri is expected to be of the form
+    dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts/
+    """
+
+    def __init__(self, artifact_uri):
+        super(DatabricksArtifactRepository, self).__init__(artifact_uri)
+        self._SERVICE_AND_METHOD_TO_INFO = {
+            service: extract_api_info_for_service(service, _PATH_PREFIX)
+            for service in [MlflowService, DatabricksMlflowArtifactsService]
+        }
+        self.credential_type_to_cloud_service = {
```

Comment:
To be deleted. I was considering a dictionary (Cloud_Provider -> upload/download_function) but it looked too messy.